### PR TITLE
WIP: Deploy xenial

### PR DESCRIPTION
**Pull request by stvstnfrd**

### CODE REVIEW

File: stanford/roles/cluster/tasks/main.yml [hunk 1]
```diff
@@ -28,28 +41,32 @@
...
-    vpc_subnet_id: "{{ CLUSTER_SUBNET_ID }}"
+    vpc_subnet_id: "{{ item.subnet.id }}"
     assign_public_ip: "{{ CLUSTER_ASSIGN_PUBLIC_IP }}"
     instance_tags:
       Birthday: "{{ lookup('pipe', \"date -u +'%Y-%m-%d %H:%M:%S'\") }}"
       Cluster: "{{ CLUSTER_NAME }}"
       Deployment: "{{ COMMON_DEPLOYMENT }}"
-      Name: "{{ COMMON_DEPLOYMENT }}-box-{{ CLUSTER_NAME }}-{{ CLUSTER_NUMBER }}"
-      Number: "{{ CLUSTER_NUMBER }}"
+      Name: "{{ COMMON_DEPLOYMENT }}-box-{{ CLUSTER_NAME }}-{{ item.item.resource_tags.Number }}"
```
Comment (stvstnfrd) [inline stanford/roles/cluster/tasks/main.yml]: TODO: Split this out so we can selectively only deploy to the first subnet.

File: stanford/lagunita/jump1.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env ansible-playbook
```
Comment (stvstnfrd) [inline stanford/lagunita/jump1.yml]: TODO: Rename play to remove `#1`.

File: stanford/stage/jump1.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env ansible-playbook
```
Comment (stvstnfrd) [inline stanford/stage/jump1.yml]: TODO: Rename to remove `#1`.

**Comment by stvstnfrd** [comment]
@caseylitton @caesar2164 @kluo @dcadams

File: stanford/playbooks/edxapp.yml [hunk 4]
```diff
@@ -1,23 +1,27 @@
 ---
-- hosts: ~tag_Name_{{ machine }}_{{ env }}
-  sudo: True
+- name: Setup edxapp
+  hosts: cluster_group_launched
+  gather_facts: True
+  become: true
+  vars:
+    CLUSTER_NAME: worker
```
Comment (stvstnfrd) [inline stanford/playbooks/edxapp.yml]: TODO: Should this be `edxapp`?

File: stanford/playbooks/forum.yml [hunk 5]
```diff
@@ -1,11 +1,48 @@
...
+    CLUSTER_NAME: forum
+    COMMON_HOSTNAME: "{{ COMMON_DEPLOYMENT }}-box-{{ CLUSTER_NAME }}-{{ CLUSTER_NUMBER }}"
   vars_files:
-    - "{{ secure_dir }}/vars/{{ env }}/forum.yml"
-    - "{{ secure_dir }}/vars/{{ env }}/users.yml"
+    - "{{ secure_dir }}/vars/{{ COMMON_DEPLOYMENT }}/forum.yml"
+    - "{{ secure_dir }}/vars/{{ COMMON_DEPLOYMENT }}/users.yml"
   roles:
+    - oraclejdk
+    - elasticsearch
```
Comment (stvstnfrd) [inline stanford/playbooks/forum.yml]: Need to be conditionalized, stage-only, I think.

File: stanford/playbooks/jump.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ansible-playbook
+---
+- include: ../playbooks/cluster.yml
+  vars:
+    ansible_ssh_common_args: ''
```
Comment (stvstnfrd) [inline stanford/playbooks/jump.yml]: TODO: Make this default and selectively override only on deploy-box.

File: stanford/playbooks/notifier.yml [hunk 7]
```diff
@@ -1,11 +1,32 @@
...
+    CLUSTER_SECURITY_GROUPS:
+      - "{{ CLUSTER_SECURITY_GROUP }}"
+- name: Create the notifier
+  hosts: cluster_group_launched
+  gather_facts: True
+  become: True
+  vars:
+    CLUSTER_NAME: certs
+    COMMON_HOSTNAME: "notifier{{ CLUSTER_NUMBER }}.{{ COMMON_DEPLOYMENT }}.vpc"
+    path_config: /opt/configuration
```
Comment (stvstnfrd) [inline stanford/playbooks/notifier.yml]: Remove this through line 24.

File: stanford/playbooks/notifier.yml [hunk 8]
```diff
@@ -1,11 +1,32 @@
...
+    repo_user: Stanford-Online
+    version_config: rc
   vars_files:
     - "{{ secure_dir }}/vars/default/common.yaml"
-    - "{{ secure_dir }}/vars/{{ env }}/notifier.yml"
-    - "{{ secure_dir }}/vars/{{ env }}/users.yml"
+    - "{{ secure_dir }}/vars/{{ COMMON_DEPLOYMENT }}/{{ CLUSTER_NAME }}.yml"
+    - "{{ secure_dir }}/vars/{{ COMMON_DEPLOYMENT }}/users.yml"
   roles:
+    # - common
```
Comment (stvstnfrd) [inline stanford/playbooks/notifier.yml]: Remove.

File: stanford/playbooks/rabbitmq.yml [hunk 9]
```diff
@@ -3,6 +3,20 @@
...
+        to_port: 5672
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 6163
+        to_port: 6163
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 15672
+        to_port: 15672
+        cidr_ip: '0.0.0.0/0'
```
Comment (stvstnfrd) [inline stanford/playbooks/rabbitmq.yml]: Add port 22 for ssh.

File: stanford/playbooks/worker.yml [hunk 10]
```diff
@@ -1,25 +1,39 @@
...
-      tags: edxapp
-      celery_worker: True
-    - role: nginx
-      tags: nginx
+- include: ../playbooks/cluster.yml
+  vars:
+    CLUSTER_INSTANCE_TYPE: t2.xlarge
+    CLUSTER_NAME: worker
+    CLUSTER_SUBNET_CIDR: '10.0.98.0/24'
+    CLUSTER_SECURITY_GROUP_RULES_INGRESS:
```
Comment (stvstnfrd) [inline stanford/playbooks/worker.yml]: TODO: Clean this up.

File: stanford/playbooks/worker.yml [hunk 11]
```diff
@@ -1,25 +1,39 @@
...
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 587
+        to_port: 587
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 6666
+        to_port: 6669
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
```
Comment (stvstnfrd) [inline stanford/playbooks/worker.yml]: Remove es?

File: stanford/playbooks/worker.yml [hunk 12]
```diff
@@ -1,25 +1,39 @@
...
+        from_port: 80
+        to_port: 80
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 587
+        to_port: 587
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 6666
```
Comment (stvstnfrd) [inline stanford/playbooks/worker.yml]: Remove IRC?

File: stanford/playbooks/worker.yml [hunk 13]
```diff
@@ -1,25 +1,39 @@
...
+        to_port: 6669
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
+        from_port: 9200
+        to_port: 9200
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
```
Comment (stvstnfrd) [inline stanford/playbooks/worker.yml]: ES?

File: stanford/playbooks/worker.yml [hunk 14]
```diff
@@ -1,25 +1,39 @@
...
+        from_port: 9300
+        to_port: 9300
+        cidr_ip: '10.0.0.0/16'
+      - proto: tcp
```
Comment (stvstnfrd) [inline stanford/playbooks/worker.yml]: Key server?

File: stanford/playbooks/xqueue.yml [hunk 15]
```diff
@@ -1,22 +1,56 @@
...
+          ping_path: '/xqueue/status/'
+          ping_port: 80
+          response_timeout: 5  # seconds
+          interval: 30  # seconds
+          unhealthy_threshold: 2  # attempts
+          healthy_threshold: 10  # attempts
+        listeners:
+          - protocol: https
+            load_balancer_port: 443
+            instance_protocol: http
```
Comment (caesar2164) [inline stanford/playbooks/xqueue.yml]: could this be the problem?

**Comment by stvstnfrd** [comment]
Test failures appear to be caused by an old version of `pip`.
I'm going to merge as-is, since things appear to function properly.

**Pull request merged**